### ISSUE TITLE
Comma outputs twice

### ISSUE THREAD

**Issue by zenite** [irssi/scripts.irssi.org#144]
Hey,
when I do:
/rsay something, and then some
the output is (from my side):
something,, and then some

**Comment by GeertHauwaerts** [comment]
Where does the /RSAY command come from?

**Comment by ailin-nemui** [comment]
either rainbow.pl or vowels.pl -- the former in this case. The bug could be in irssi as well, in case mIRC parses colour codes as always removing the trailing `,`. Someone should check that

**Comment by vague666** [comment]
In rainbow.pl
        for (my $c = 0; $c < length($string); $c++) {
                my $char = substr($string, $c, 1);
                if ($char eq ' ') {
                        $newstr .= $char;
                        next;
                }
                while (($color = int(rand(scalar(@colors)))) == $last) {};
                $last = $color;
                $newstr .= "\003";
                $newstr .= sprintf("%02d", $colors[$color]);
                $newstr .= (($char eq ",") ? ",," : $char);
        }

simple test by changing ",," => "," shows no errors, not sure what ",," is supposed to accomplish, maybe meant to disable a possible background color?

No visible change in mIRC, shows double ',' when present in script, shows single ',' when script is modified

![mirc](https://cloud.githubusercontent.com/assets/8029742/11399797/e7057c88-9389-11e5-81a4-72697e4d972a.png)

**Comment by ailin-nemui** [comment]
thanks for checking, sounds like an easy fix then

**Comment by vague666** [comment]
Yeah, shall I push the change to the script?

**Comment by ailin-nemui** [comment]
sure

**Comment by dequis** [comment]
I dunno, that line looks very intentional, just removing it without understanding why it's there seems wrong.

**Comment by vague666** [comment]
It does look intentional but the author should comment on it in that case, my tests show no problems with removing it from the script though.

**Comment by ailin-nemui** [comment]
Thanks for cross-checking, the issue that the script creator was probably having in mind is: if you have a text like `X,00` and you insert `X<C>05,00` then the background colour would change and the comma would disappear. But it should not be an issue in the rainbow script anyway, as the script goes on and modifies the text further like `X<C>05,<C>040<C>030...`

**Comment by ailin-nemui** [comment]
The other issue that he could have in mind is when colour-code parsers eat the comma after a colour code, because they believe it to be part of the colour specification. But apparently this is not the case

**Comment by dequis** [comment]
Double bold?

```diff
-$newstr .= (($char eq ",") ? ",," : $char);
+$newstr .= (($char eq ",") ? "\002\002," : $char);
```

**Comment by ailin-nemui** [comment]
well according to vague's experiments this doesn't seem to be any actual issue so we don't to do anything special (contrary to what the author implemented)

**Comment by vague666** [comment]
Maybe it was a problem when the script was written?

**Comment by ailin-nemui** [comment]
could have been a bug in irssi at that time... https://github.com/irssi/irssi/commit/3f0743e749e1b0bce6f7a7deb304643367f90252

**Comment by dequis** [comment]
Plausible:

```
# Written by Jakub Jankowski <shasta@atn.pl>
# for Irssi 0.7.98.4 and newer
```

I didn't even know that version existed.

Merging vague's PR as-is sounds better now.

**Comment by ailin-nemui** [comment]
Fixed via #184
